dockertest/subtestbase.py

```diff
@@ -102,10 +161,84 @@ def cleanup(self):
         Always called, before any exceptions thrown are re-raised.
         """
         self.log_step_msg('cleanup')
+        self.overlook_known_failures()
+
+    def overlook_known_failures(self):
+        """
+        If we've caught a DockerXxxx exception, see if it's a known problem
+        with this combination of subtest & docker NVR. If it is, change the
+        exception status from FAIL to WARN.
+        """
+        e_type, e_value, _ = sys.exc_info()
+
+        # A string comparison lets us catch any DockerXxxxx exception,
+        # e.g. DockerTestFail, DockerOutputError
+        if 'Docker' not in str(e_type):
```

Comment:
This should probably be ``isinstance(e_type, xceptions.whatever)`` no?

Reply:
Grrrrr. I knew this would be complicated. I even had a long comment there but it grew beyond any hope of sanity. Yes, `isinstance()` is how I started. I gave up on it.

First: for reasons I still don't understand, sometimes a test throws `xceptions.DockerTestFail` and sometimes `dockertest.xceptions.DockerTestFail`; and `isinstance()` isn't smart enough to realize that both are the same. I would have to test for both.

Second: it's not just `DockerTestFail`. I've also seen `DockerOutputError`. I realize that a string compare is ugly but the alternative I see is a long and fragile enumeration of exceptions. Maybe there's a cleaner way that I've missed.

Reply:
>  I would have to test for both.

Ooohh right, I remember that strangeness now.  We ran into that in another lifetime (PR) w/o any conclusion.  I can live with what you have, but surely if I can forget, anybody can.  Maybe make a simple comment about this as a reminder.  How about:

```python
# xceptions.DockerXYZ makes isinstance() mad sometimes, cheat by looking at the name
```

?